### ISSUE TITLE
How to understand sample from softmax?

### ISSUE THREAD

**Issue by npuichigo** [pclucas14/pixel-cnn#2]
I have no idea of this line about sampling from softmax. Can you give me a help?
```
 temp.uniform_(1e-5, 1. - 1e-5)
 temp = logit_probs.data - torch.log(- torch.log(temp))
 _, argmax = temp.max(dim=3)
```

**Comment by pclucas14** [comment]
Hi @npuichigo 

Sorry for the late reply; I took this line from the original repository. From what I understand it has to do with the definition of the CDF for the mixture of logistic distribution.
